pywr/domains/river.py

```diff
@@ -298,3 +467,94 @@ def fset(self, value):
         return locals()
 
     cost = property(**cost())
+
+class ProportionalInput(Input, metaclass=NodeMeta):
+    """
+    TODO: Jose explain this
+    """
+    min_proportion = 1e-6
+
+    def __init__(self, model, name, node, proportion, **kwargs):
+        super().__init__(model, name, **kwargs)
+
+        self.node = model.pre_load_node(node)
+
+        # Create the flow factors for the other node and self
+        if proportion < self.__class__.min_proportion:
+            self.max_flow = 0.0
+        else:
+            factors = [1, proportion]
+            # Create the aggregated node to apply the factors.
+            self.aggregated_node = AggregatedNode(model, f'{name}.aggregated', [self.node, self])
+            self.aggregated_node.factors = factors
+
+
+
+class LinearStorageReleaseControl(Link, metaclass=NodeMeta):
```

Comment:
I wonder if this should be a seperate merge request. A linear release rule is a very special case, that I've only seen used in teaching. Real rules tend to be more complex. I think we need to think about this more on how to implement reservoir release rules intuitively. Also the cost shouldn't always be 0. This woudl work because there was a turbine downstream with a negative cost normally when these teaching models. 

In the most basic form, a release rule node is a normal pywr node with a max flow that is the release rule (which can be a constant, ControlCurveInterpolatedParameter, radialbasis function, monthlyparam or any number of Pywr parameters and a negative cost which should be a value that makes sense based on the system